src/git.rs

```diff
@@ -50,18 +59,77 @@ pub fn checkout(repo: &Repository, commit_id: &str) -> Result<(), Error> {
     repo.reset(&obj, Hard, Some(&mut checkout_opts))
 }
 
-pub fn checkout_submodules(repo: &Repository) -> Result<(), Error> {
-    for mut submodule in repo.submodules()? {
-        info!("getting submodule {:?} from {:?}", submodule.name(), submodule.url());
-        submodule.update(true, None)?
+fn clone<P>(
+    repository_url: &str,
+    into_dir: P,
+    get_credentials: &impl Fn() -> Vec<(CredentialType, Cred)>,
+) -> Result<Repository, Error>
+where
+    P: AsRef<Path>,
+{
+    let url = Url::parse(repository_url)
+        .map_err(|err| Error::from_str(format!("Invalid repository url {}: {}", repository_url, err).as_str()))?;
+
+    if url.scheme() != "https" {
+        return Err(Error::from_str("Repository URL have to start with https://"));
```

Comment:
Isn't it a bit weird to only allow ssh for submodule but not main repo ?

Reply:
(This line verify that it is https and not just http, as we send credentials along)

For your general question,
For submodule, you can clone them via https:// (modulo it is public or that your credentials works on it) or via ssh (but only with ssh key, as it is the only auth method possible with this scheme)

For cloning the main repo, we allow only `https://` just to avoid mistake, credentials passed in the git_reposiory object should be valid and as login/pass with auth token is only possible with httpx scheme, I force it

Reply:
Ok got it ! Thx